api/src/main/java/org/openmrs/module/appointments/events/eventListener/AppointmentSMSEventListener.java

```diff
@@ -0,0 +1,97 @@
+    @EventListener
+    public void onApplicationEvent(AppointmentBookingEvent event) {
+        try {
+            Context.openSession();
+            Context.setUserContext(event.userContext);
+            if (event.eventType == AppointmentEventType.BAHMNI_APPOINTMENT_CREATED) {
+                handleAppointmentCreatedEvent(event.getAppointment());
+            }
+        } catch (Exception e) {
+            log.error("Exception occurred during event processing", e);
+        } finally {
+            Context.closeSession();
+        }
+    }
+
+    @Async("AppointmentsAsyncThreadExecutor")
+    @EventListener
+    public void onApplicationEvent(RecurringAppointmentEvent event) {
+        try {
+            Context.openSession();
+            Context.setUserContext(event.userContext);
+            if (event.eventType == AppointmentEventType.BAHMNI_RECURRING_APPOINTMENT_CREATED) {
+                handleRecurringAppointmentCreatedEvent(event.getAppointmentRecurringPattern().getAppointments().iterator().next());
+            }
+        } catch (Exception e) {
+            log.error("Exception occurred during event processing", e);
+        } finally {
+            Context.closeSession();
+        }
+    }
+
+    private void handleAppointmentCreatedEvent(Appointment appointment) {
+        String phoneNumber=getPhoneNumber(appointment.getPatient());
+        if (!checkGlobalCondition()||phoneNumber==null)return;
+        MessageBuilderService messageBuilderService = Context.getRegisteredComponent("messageBuilderService", MessageBuilderService.class);
+        CommunicationService communicationService = Context.getRegisteredComponent("communicationService", CommunicationService.class);
+        List<String> providersName = appointmentArgumentsMapper.getProvidersNameInString(appointment);
+        String message = messageBuilderService.getAppointmentBookingMessage(appointmentArgumentsMapper.createArgumentsMapForAppointmentBooking(appointment), providersName);
+        communicationService.sendSMS(phoneNumber, message);
+    }
+
+    private void handleRecurringAppointmentCreatedEvent(Appointment appointment) {
+        String phoneNumber=getPhoneNumber(appointment.getPatient());
+        if (!checkGlobalCondition()||phoneNumber==null)return;
+        MessageBuilderService messageBuilderService = Context.getRegisteredComponent("messageBuilderService", MessageBuilderService.class);
+        CommunicationService communicationService = Context.getRegisteredComponent("communicationService", CommunicationService.class);
+        List<String> providersName = appointmentArgumentsMapper.getProvidersNameInString(appointment);
+        String message = messageBuilderService.getAppointmentBookingMessage(appointmentArgumentsMapper.createArgumentsMapForRecurringAppointmentBooking(appointment), providersName);
+        communicationService.sendSMS(phoneNumber, message);
+    }
+    private boolean checkGlobalCondition() {
```

Comment:
may a method name "shouldSendEmailForBookingAppointment()" is more appropriate than "checkGlobalCondition"?